test/src/unit-cppapi-global-order-writes-remote.cc

```diff
@@ -0,0 +1,504 @@
+/**
+ * @file unit-cppapi-global-order-writes-remote.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2023 TileDB Inc.
+ * @copyright Copyright (c) 2016 MIT and Intel Corporation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * Tests for global order remote writes.
+ */
+
+#include <test/support/tdb_catch.h>
+#include "tiledb/sm/cpp_api/tiledb"
+
+#include <cstring>
+#include <iostream>
+
+using namespace tiledb;
+
+const std::string aws_region = "us-east-2";
+const std::string aws_key_id = "<AWS_KEY_ID>";
+const std::string aws_secret_key = "<AWS_SECRET_KEY>";
+
+// Toggle remote queries for sanity check.
```

Comment:
I would remove the keys above and everything related to this flag. You can make a python script for yourself to test this separately and it shouldn't be committed in our unit tests. Also includes the user name/password you use for tests...

Reply:
If I remove the remote portion of these tests then I would be writing to local arrays.. I can do this, but the tests wouldn't be testing remote global order writes anymore. :confused:  

The username and password here is arbitrary, so for now I just changed it to `demo` and `demodemo`. It's the username and password I have configured on my local REST server for testing. I was able to get to most of your feedback today, aside from this one and the `query_` member of `QueryRemoteBufferStorage`. I'll take another look at these in the morning. :eyes: